vllm/config.py

```diff
@@ -1438,55 +1440,63 @@ def __post_init__(self):
 class ParallelConfig:
     """Configuration for the distributed execution."""
 
-    pipeline_parallel_size: int = 1  # Number of pipeline parallel groups.
-    tensor_parallel_size: int = 1  # Number of tensor parallel groups.
-    data_parallel_size: int = 1  # Number of data parallel groups.
-    data_parallel_rank: int = 0  # Rank of the data parallel group.
-    # Local rank of the data parallel group, defaults to global rank.
+    pipeline_parallel_size: int = 1
```

Comment:
One thing that's not clear from the code is whether `pipeline_parallel_size` (and the rest) are intended to be constants that always refer to the default values, or if they may change at runtime.

If they may change, then something like this would be nice to make it more explicit so that it's also clear when referencing these values from other parts of the code:

```suggestion
    DEFAULT_PIPELINE_PARALLEL_SIZE: int = 1
    pipeline_parallel_size: int = DEFAULT_PIPELINE_PARALLEL_SIZE
```

Reply:
When they are referred to in `EngineArgs` and the argparser defaults we refer to the class attributes (`ParallelConfig.pipeline_parallel_size`), not instance attributes (`ParallelConfig().pipeline_parallel_size`). Therefore, they cannot change at runtime (without modifying the `ParallelConfig` class itself) and will always be constants that refer to the default.

Adding extra attributes to the dataclass that define defaults doesn't seem very Pythonic.

Reply:
I missed/forgot this was a `dataclass` and not a normal class, so the behavior is already what I was going for - nevermind!